iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm

```diff
@@ -640,6 +667,41 @@ - (void)updateBookmark:(MWMMarkID)bookmarkId
   }
 }
 
+- (void)moveBookmark:(MWMMarkID)bookmarkId
+           toGroupId:(MWMMarkGroupID)groupId {
+    auto const currentGroupId = self.bm.GetBookmark(bookmarkId)->GetGroupId();
+    auto editSession = self.bm.GetEditSession();
+    if (groupId != kml::kInvalidMarkGroupId) {
+        editSession.MoveBookmark(bookmarkId, currentGroupId, groupId);
+    }
+}
+
+- (void)updateTrack:(MWMTrackID)trackId
+         setGroupId:(MWMMarkGroupID)groupId
+              title:(NSString *)title {
+    auto const currentGroupId = self.bm.GetTrack(trackId)->GetGroupId();
+    auto editSession = self.bm.GetEditSession();
+    if (groupId != kml::kInvalidMarkGroupId) {
+        editSession.MoveTrack(trackId, currentGroupId, groupId);
+    }
+
+    auto track = editSession.GetTrackForEdit(trackId);
+    if (!track) {
+        return;
+    }
+    
+    track->SetName(title.UTF8String);
```

Comment:
It may be better to split move and edit into separate functions.

Reply:
I was planned to update colors here too, so the function would be `updateTrack:setGroupId:title:color:` (and also update track line with too if I'll find the way).
Track move is a part of track edit (if user changes group too).

This function is similar to `updateBookmark:...` which moves bookmark, changes title, description, color at once.
Should we make `updateTrack` different from `updateBookmark`?